### PR TITLE
cp: -i prompts in the right place and other improv

### PR DESCRIPTION
**Pull request by sylvestre**

### CODE REVIEW

File: tests/by-util/test_cp.rs [hunk 1]
```diff
@@ -444,16 +444,26 @@ fn test_cp_arg_interactive() {
...
     at.touch("a");
     at.touch("b");
-    ucmd.args(&["-i", "-u", "a", "b"])
+    ucmd.args(&["-vi", "a", "b"])
         .pipe_in("N\n")
-        .succeeds()
-        .no_stdout();
+        .fails()
+        .stdout_is("skipped 'b'\n");
+}
```
Comment (cakebaker) [inline tests/by-util/test_cp.rs:455]: What's the reason for replacing the `-iu`-test with a `-vi`-test?

Comment (sylvestre) [reply tests/by-util/test_cp.rs:455]: typo, let me fix that :)

File: tests/by-util/test_cp.rs [hunk 2]
```diff
@@ -456,6 +456,30 @@ fn test_cp_arg_interactive_update() {
...
+
+#[test]
+#[cfg(not(target_os = "android"))]
+fn test_cp_arg_interactive_verbose_clobber() {
+    let (at, mut ucmd) = at_and_ucmd!();
+    at.touch("a");
+    at.touch("b");
+    ucmd.args(&["-vin", "a", "b"])
+        .pipe_in("y\n")
```
Comment (cakebaker) [inline tests/by-util/test_cp.rs:478]: As there is no prompt in this test case, it's not relevant what you pipe in, so `pipe_in` is not needed.
```suggestion
```

File: src/uu/cp/src/cp.rs [hunk 3]
```diff
@@ -1107,18 +1108,20 @@ fn show_error_if_needed(error: &Error) -> bool {
...
-            return true;
+            true
         }
         _ => {
             show_error!("{}", error);
-            return true;
+            true
         }
     }
-    false
```
Comment (cakebaker) [inline src/uu/cp/src/cp.rs:1121]: I would remove the entire function because now it always returns `true`.

**Review by cakebaker — APPROVED** [review]

**Pull request merged**